services/github/github-search.service.js

```diff
@@ -7,9 +7,66 @@ import { documentation } from './github-helpers.js'
 
 const schema = Joi.object({ total_count: nonNegativeInteger }).required()
 
-export default class GithubSearch extends GithubAuthV3Service {
+const codeSearchDocs = `
+For a full list of available filters and allowed values,
+see GitHub's documentation on
+[Searching code](https://docs.github.com/en/search-github/github-code-search/understanding-github-code-search-syntax)
+
+${documentation}`
+
+class BaseGithubCodeSearch extends GithubAuthV3Service {
   static category = 'analysis'
 
+  static defaultBadgeData = {
+    label: 'counter',
+  }
+
+  static render({ query, totalCount }) {
+    return {
+      label: `${query} counter`,
+      message: metric(totalCount),
+      color: 'blue',
+    }
+  }
+}
+
+class GitHubCodeSearch extends BaseGithubCodeSearch {
+  static route = {
+    base: 'github/search',
+    pattern: ':query+',
+  }
```

Comment:
Good catch on this issue.

In terms of how to solve it..

There's no way to distinguish between `/:user/:repo/:query+` and a `/:query+` that contains multiple slash characters.

One way round this would be to put the non repo-scoped version on a different route (so we pick `/github/global-search/:query+` or something?). However, I think the optimal way to solve this would be to match what we've done with the GitHub Issue Search badge and make `query` a QueryString param instead of a path param.

https://shields.io/badges/git-hub-issue-custom-search
https://shields.io/badges/git-hub-issue-custom-search-in-repo

That way they could be `/github/search?query=foo` and `/github/search/user/repo?query=foo`, and `query` can contain any number of slashes. It would also make the URL format match up with this existing conceptually similar badge.

In order to do this, we'd also have to add a redirector for `/github/search/:user/:repo/:query+` which converts `query` to a QueryString param to avoid breaking existing badges. There is an example of how to do this at https://github.com/badges/shields/blob/master/services/gitlab/gitlab-coverage-redirect.service.js - this redirect switches branch to a query param.

Reply:
Thank you for your actionable feedbacks. I'll take a look. 

Stay tuned

Reply:
OK guys, I have refactored.

The code works BUT I'm unsure the redirection does.
I tried multiple things, the tests are passing, but I wasn't able to see it working.

Please review, and tell me if there is something obvious I missed

Reply:
@jNullj @chris48s ? maybe

Reply:
For records, here is the "reason" why you were not able to push to "my fork"

- https://github.com/ccoVeille/github-forks-sanitizer/issues/3#issue-2729309577